db/index.go

```diff
@@ -290,10 +290,16 @@ func (i *collectionUniqueIndex) newUniqueIndexError(
 	doc *client.Document,
 ) error {
 	fieldVal, err := doc.GetValue(i.fieldDesc.Name)
+	var val any
 	if err != nil {
-		return err
+		if !errors.Is(err, client.ErrFieldNotExist) {
```

Comment:
todo: Can you please document why this if exists, it will not be obvious when reading the code at a later date.